src/algorand/AlgorandEngine.ts

```diff
@@ -95,6 +106,152 @@ export class AlgorandEngine extends CurrencyEngine<
+                  connector,
+                  wcProps,
+                  dApp
+                }
+                resolve(dApp)
+              }
+            )
+
+            // Subscribe to call requests
+            connector.on(
+              'call_request',
+              (error: Error | null, payload: AlgoWcRpcPayload) => {
+                try {
+                  if (error != null) throw error
+                  const cleanPayload = asAlgorandWalletConnectPayload(payload)
+                  const params = cleanPayload.params[0][0]
+                  const algoTx = decodeUnsignedTransaction(
+                    base64.parse(params.txn)
+                  )
+
+                  const nativeAmount =
+                    algoTx.amount != null ? algoTx.amount.toString() : '0'
+                  const networkFee = algoTx.fee.toString()
+                  let tokenId: string | undefined
+
+                  if (algoTx.type === 'axfer') {
+                    const assetIndex = algoTx.assetIndex.toString()
+                    const metaToken: EdgeToken | undefined =
+                      this.allTokensMap[assetIndex]
+                    if (metaToken == null) throw new Error('Unrecognized token')
+                    tokenId = assetIndex
+                  }
+
+                  const out = {
+                    uri: wcProps.uri,
+                    dApp: this.tools.walletConnectors[wcProps.uri].dApp,
+                    payload: cleanPayload,
+                    walletId: this.tools.walletConnectors[wcProps.uri].walletId,
+                    nativeAmount,
+                    networkFee,
+                    tokenId // optional
+                  }
+
+                  this.currencyEngineCallbacks.onWcNewContractCall(out)
+                } catch (e: any) {
+                  this.warn(`Wallet connect call_request `, e)
+                  throw e
+                }
+              }
+            )
+
+            // eslint-disable-next-line @typescript-eslint/no-unused-vars
+            connector.on('disconnect', (error, payload) => {
+              if (error != null) {
+                throw error
+              }
+              // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
+              delete this.tools.walletConnectors[wcProps.uri]
+            })
+          }),
+          5000
+        )
+      },
+      wcConnect: (uri: string, publicKey: string, walletId: string) => {
+        this.tools.walletConnectors[uri].connector.approveSession({
+          accounts: [publicKey],
+          chainId: 416001 // Not actually required in runtime. Using the placeholder value found in another implementation.
+        })
+        this.tools.walletConnectors[uri].walletId = walletId
+      },
+      wcDisconnect: (uri: string) => {
+        // eslint-disable-next-line @typescript-eslint/no-floating-promises
+        this.tools.walletConnectors[uri].connector.killSession()
```

Comment:
`.catch(()=>{})` would remove the need for the eslint-disable, and would make it clear that we don't care about the return value at all.